packages/ilmomasiina-backend/src/routes/signup/getSignupForEdit.ts

```diff
@@ -0,0 +1,71 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+import { NotFound } from 'http-errors';
+
+import * as schema from '@tietokilta/ilmomasiina-models/src/schema';
+import { Answer } from '../../models/answer';
+import { Event } from '../../models/event';
+import { Question } from '../../models/question';
+import { Quota } from '../../models/quota';
+import { Signup } from '../../models/signup';
+import { nullsToUndef, stringifyDates } from '../utils';
+
+// TODO: Require editTokenVerification
+export default async function getSignupForEdit(
+  request: FastifyRequest<{ Params: schema.SignupPathParams }>,
+  reply: FastifyReply,
+): Promise<schema.UserSignupForEditSchema> {
+  const signup = await Signup.scope('active').findByPk(request.params.id, {
+    include: [
+      {
+        model: Answer,
+        required: false,
+      },
+      {
+        model: Quota,
+        include: [
+          {
+            model: Event,
+            include: [
+              {
+                model: Question,
+                required: false,
+              },
+            ],
+          },
+        ],
+      },
+    ],
+    order: [[Quota, Event, Question, 'order', 'ASC']],
+  });
+  if (signup === null) {
+    // Event not found with id, probably deleted
+    throw new NotFound('No signup found with given id');
+  }
+
+  const event = signup.quota!.event!;
+
+  const response = {
+    signup: {
+      ...stringifyDates(signup.get({ plain: true })),
+      status: signup.status,
+      firstName: nullsToUndef(signup.firstName),
+      lastName: nullsToUndef(signup.lastName),
+      email: nullsToUndef(signup.email),
+      confirmed: !!signup.confirmedAt,
+      answers: signup.answers!,
+      quota: signup.quota!,
+    },
+    event: {
+      ...stringifyDates(event.get({ plain: true })),
+      questions: event.questions!.map((question) => ({
+        ...question.get({ plain: true }),
+        // Split answer options into array
+        options: question.options ? question.options.split(';') : null,
+      })),
+    },
+  };
```

Comment:
What's the purpose of `nullsToUndef` here?

```suggestion
  const response = stringifyDates({
    signup: {
      ...signup.get({ plain: true }),
      status: signup.status,
      firstName: nullsToUndef(signup.firstName),
      lastName: nullsToUndef(signup.lastName),
      email: nullsToUndef(signup.email),
      confirmed: !!signup.confirmedAt,
      answers: signup.answers!,
      quota: signup.quota!,
    },
    event: {
      ...event.get({ plain: true }),
      questions: event.questions!.map((question) => ({
        ...question.get({ plain: true }),
        // Split answer options into array
        options: question.options ? question.options.split(';') : null,
      })),
    },
  });
```

Reply:
The purpose of `nullsToUndef`:
In SQL, natural way to present non-existing optional data is to use `NULL`. In JSON, however, null values won't look that great. The more native way of describing an non-existing optional data is to leave such keys undefined (and not present in the serialized JSON).

Reply:
For the suggestion of combining the two `stringifyDates`, I would rather not combine them; the penalty from repetition is not that big in this specific scenario, using such powerful recursive functions in larger scope can accidentally touch to somewhere they shouldn't and also runtime performance is slightly better as the calls are better scoped. TS enforces using types matching with the schema.